prog/vm/github_runner.rb

```diff
@@ -220,6 +220,71 @@ def setup_info
     # Remove comments and empty lines before sending them to the machine
     vm.sshable.cmd(command.gsub(/^(\s*# .*)?\n/, ""))
 
+    if github_runner.installation.project.get_ff_transparent_cache
+      hop_setup_forked_runner
+    else
+      hop_register_runner
+    end
+  end
+
+  label def setup_forked_runner
+    command = <<~COMMAND
+      if [ ! -f "actions-runner.tar.gz" ]; then
+        curl --output actions-runner.tar.gz -L #{Config.github_cache_forked_runner_tarball_uri}
+      fi
+
+      mkdir -p actions-runner
+      if [ ! -d "actions-runner/bin" ]; then
+        tar xzf actions-runner.tar.gz -C actions-runner
+      fi
+
+      sudo chown -R runneradmin:runneradmin ./actions-runner
+
+      cat <<EOT > ./actions-runner/run-withenv.sh
+      #!/bin/bash
+      mapfile -t env </etc/environment
+      exec env -- "${env[@]}" ./actions-runner/run.sh --jitconfig "$1"
+      EOT
+      chmod +x ./actions-runner/run-withenv.sh
+
+      jq '. += [#{setup_info.to_json}]' /imagegeneration/imagedata.json | sudo -u runner tee ./actions-runner/.setup_info
+
+      echo "PATH=$PATH" >> ./actions-runner/.env
+
+      rm -rf /home/runner/actions-runner
+      sudo mv ./actions-runner /home/runner/
+      sudo chown -R runner:runner /home/runner/actions-runner
```

Comment:
What do you think about directly extracting to `/home/runner/actions-runner` using the `--overwrite` flag?

```suggestion
      sudo tar xzf actions-runner.tar.gz -C /home/runner/actions-runner --overwrite
```

Reply:
I've tried that but it couldn't set the permissions properly. Also, I wanted to be more explicit, so keeping it as it is.